src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -111,11 +116,23 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
           this.props.onSelectedDataSources([{ id: optionId, label: selectedDataSource.title }]);
         }
       } catch (error) {
-        handleDataSourceFetchError(
-          this.onError.bind(this),
-          this.props.notifications!,
-          this.props.onSelectedDataSources
-        );
+        // return [] to customer when no default data source
+        if (!defaultDataSource && this.props.hideLocalCluster) {
+          this.props.onSelectedDataSources?.([]);
+        }
+        const filteredDefaultDataSource = this.getFilteredDefaultDataSource(defaultDataSource);
+        this.setState({
+          defaultDataSource: filteredDefaultDataSource,
```

Comment:
can we avoid setstate multiple times in one block?

Reply:
updated